### DISCUSSION TITLE
`onUploadProgress` doesn't work when `data` is a stream

### DISCUSSION THREAD

**Discussion by strarsis** [axios/axios#4952] [General]
Related: https://github.com/axios/axios/issues/961

So I use a stream (from a file) as `data` in an Axios request in order to `POST` the file bytes as request body (required by that endpoint), so no multipart/form body.

This works fine now in recent Axios.

However, the `onUploadProgress` is never called. The reason for this is probably that a stream is used, and not a buffer or `FormData` with the file added to it.
As those files are larger, it would be nice to stream them anyway.

The `content-length` and `Content-Length` headers are correctly set as Axios request header.

**Comment by jasonsaayman** [comment]
Can you elaborate on an issue and tag me?

**Comment by strarsis** [comment]
@jasonsaayman:
```js
const MY_FILE_PATH = './image.jpg';

const {size} = fs.statSync(MY_FILE_PATH);
const fileStream = fs.createReadStream(MY_FILE_PATH);

const upStream = await axios({
        method: 'POST',
        url: 'https://example.com/upload',
        headers: {
            'Content-Type': 'image/jpeg',
            'Content-Length': size,
        },
        data: fileStream,
        onUploadProgress: progressEvent => console.log(progressEvent.loaded)
    })
````

The `onUploadProgress` handler function is not called. The promise resolves when the upload finished.

Edit: On an issue, alright, I create one.

**Comment by DigitalBrainJS** [comment]
This should work in Axios v1.0.0. Previously, Axios did not support progress capturing in the server environment, only for browser platforms.
As far as I remember, progress capture for FormData in the node.js environment is not yet supported. There are architectural limitations in the current implementation of the Axios core that prevent this from being added for now. However, capturing the upload/download progress of regular streams should work if the content-size header is specified.
P.S For streams always set maxRedirects to 0 to avoid memory leaks.
